swan-lake/resources/featured-scenarios/create-your-first-connector-with-ballerina.md

```diff
@@ -0,0 +1,166 @@
+
+## Step 2: Prepare the OpenAPI Specification
+
+1. Find the OpenAPI specification for the API you want to create a connector for. This is usually available in the API documentation.
+   Example: For Twitter, you can get the latest API specification from the [Twitter OpenAPI endpoint](https://api.twitter.com/2/openapi.json).
+
+2. Save the file as `spec.yaml` (or `spec.json`) in the `docs/spec` directory of your project.
+
+3. You may need to sanitize the OpenAPI specification to ensure compatibility with the Ballerina OpenAPI tool. This process may involve:
+   - Renaming schema names to comply with Ballerina naming conventions.
+   - Adding, removing, or modifying security schemes to customize authentication options.
+
+## Step 3: Generate the Ballerina client code
+
+With your OpenAPI spec ready, use the [Ballerina OpenAPI tool](https://ballerina.io/learn/openapi-tool/) to generate the connector code.
+
+1. In your terminal, run the following command from the project root:
+   ```
+   bal openapi -i path/to/spec --mode client -o ballerina
+   ```
+
+   This will generate the ballerina source code of the connector client in your `ballerina/` directory.
+
+> **Note:** Ballerina OpenAPI tool supports multiple customization options when generating clients. For more details, check the [Ballerina OpenAPI tool documentation](https://ballerina.io/learn/openapi-tool/).
+
+2. Make sure that the generated client consists of the following files:
+   - `client.bal`: Contains the client implementation with all the API operations.
+   - `types.bal`: Contains the data types used in the client.
+   - `utils.bal`: Contains utility functions used in the client.
+
+## Step 4: Test the connector
+
+Now that your connector is generated, it’s important to write tests to ensure everything works as expected.
+
+1. In the `ballerina/tests` directory, add the required test files and write test cases for the key operations of your connector. Aim to cover as many API use cases as possible.
+
+2. Execute the tests using the `bal test` command to verify the functionality of your connector.
+
+> For detailed guidance on writing tests, check the [Ballerina testing guide](https://ballerina.io/learn/test-ballerina-code/test-a-simple-function/).
+
+> For sample implementations, refer to the [test implementation of the Ballerina Twitter connector](https://github.com/ballerina-platform/module-ballerinax-twitter/tree/main/ballerina/tests).
+
+## Step 5: Document the connector
+
+Follow these steps to ensure your connector is well-documented:
+
+1. Update `Module.md` and `Package.md` files
+
+These files will be displayed on the Ballerina Central package landing page. Ensure they introduce and explain the connector package clearly and comprehensively. Each file should include the following sections:
```

Comment:
May have to be indented.

Reply:
Updated in https://github.com/ballerina-platform/ballerina-dev-website/pull/9434/commits/7b6290006703ddda90e785b37d5a853c1b3f773d